CellProfiler_pipelines/README.md

```diff
@@ -0,0 +1,107 @@
+There are three sets of `CorrectIlluminationCalculate` and `CorrectIlluminationApply` modules, where each set corrects a different channel and each contains different set of parameters as I have found using the same parameters for each channel does not always work.
+
+For more information regarding these modules, you can reference my [Illumination Correction prototyping README](https://github.com/WayScience/CellProfiler_Prototyping/tree/main/1.illumination_correction) within the [CellProfiler_Prototyping repo](https://github.com/WayScience/CellProfiler_Prototyping). 
+This README describes each of the parameters within these modules and provides further references.
+
+Since the NF1 pilot data is small, this IC pipeline saves all of the images to use in the next pipeline.
+For much larger datasets, it is standard to save the illumination correction function (which is much smaller and saves storage) and use that to perform IC in the next pipeline.
+
+## Analysis Pipeline
+
+In the [NF1_analysis.cpproj](Pipelines/NF1_analysis.cpproj) CP project, I perform segmentation and feature extraction.
+
+### Segmentation
+
+To perform segmentation in CP, the modules that were used were:
+
+- `IdentifyPrimaryObjects` Identify nuclei from the DAPI channel images. This creates a group for an object called `OrigNuclei`, which includes nuclei that are a part of cells that touch an edge of the image.
+
+- `IdentifySecondaryObjects` Identify whole cells using the nuclei from the previous module as a base from the RFP channel (stained for actin) images. 
+These whole cells make up the object group called `Cells`.
+Any cells that are touching at least one edge of the image will be removed along with its respective nucleus. 
+This creates a new object group call `Nuclei`.
+
+- `IdentifyTeritaryObjects` Identify cytoplasm by subtracting out the "smaller identfied objects" (`OrigNuclei`) from the "larger identified objects" (`Cells`). 
+`OrigNuclei` has to be used in this module over `Nuclei` due to an error downstream in the pipeline when using the latter object where it states "Cytoplasm is not in a 1:1 relationship with other objects".
+This means that the number of cytoplasm is not the same as the other objects, which would be wrong because there should be the same amount of objects (e.g the number of cells should match the number of cytoplasm and nuclei).
+
+Based on my understanding of CellProfiler, these modules identify the objects and create masks for each of them to use for feature extraction.
+
+### Feature Extraction
+
+By using the object masks, each module runs through each image from every channel to extract features from.
+I followed the format of one of the CP pipelines from the [Broad Institute Platform Pipelines](https://github.com/broadinstitute/imaging-platform-pipelines/blob/master/cellpainting_UMUC9_20x_phenix_bin2/analysis_without_batchfile.cppipe).
+The following modules were used:
+
+- `MeasureColocalization`
```

Comment:
Nice. I like how you provided rationale for these choices.

A couple of discussion items:

1. Do these modules have additional parameters that you needed to tweak?
2. If we decided to add a different module (e.g. `MeasureObjectIntensity`) would it be as straightforward as I think it would be? Or would it be more nuanced to add new modules?

Reply:
It is super easy to add new modules! Actually, I should have at that module in there but I guess when I made it I missed it. 

@gwaybio I will add that module here on my computer at work, rerun the analysis and get a new sqlite file.

Also, there are parameters in some of them to tweak. I can add them, but documentation on what they do are provide under the `?` button in CP. @gwaybio Do you want me to add a brief description of the parameter and what it does or let them use the provided help button?

Reply:
Our goal is to provide enough documentation such that a reader understands our process.

The `cpipe` (+ instructions on how to run) is sufficient for computational reproducibility - our goal with this documentation is understanding.

Please do what you feel is the right thing to do given our goals.

Reply:
I will add brief descriptions then of what I understand of these parameters. Ty!